homework18/src/main/java/Box.java

```diff
@@ -0,0 +1,46 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Box<T extends Fruit> {
+
+    List<T> fruits;
+
+    public Box() {
+        fruits = new ArrayList<>();
+    }
+
+    public void addFruit(T fruit) {
+        fruits.add(fruit);
+    }
+
+    public void addFruits(List<T> fruits) {
+        this.fruits.addAll(fruits);
+    }
```

Comment:
You should check each fruit because you're not allowed to add fruit of a different type.